pkg/api/validation/validation.go

```diff
@@ -1276,17 +1276,51 @@ func validateEndpointSubsets(subsets []api.EndpointSubset) errs.ValidationErrorL
 		if len(ss.Ports) == 0 {
 			ssErrs = append(ssErrs, errs.NewFieldRequired("ports"))
 		}
-		// TODO: validate each address and port
+		for addr := range ss.Addresses {
+			ssErrs = append(ssErrs, validateEndpointAddress(&ss.Addresses[addr])...)
```

Comment:
This still looks like it is going to come out with the wrong field name.  It's pretty ugly, but you have to do something like PrefixIndex(addr).Prefix("addresses") - wee how other array validations are handled.  We're not always consistent, but we should try to be.